solr-ingest/src/main/java/edu/unc/lib/dl/data/ingest/solr/filter/SetAccessStatusFilter.java

```diff
@@ -105,13 +108,45 @@ public void setObjectAclFactory(ObjectAclFactory oaf) {
             status.add(FacetConstants.PUBLIC_ACCESS);
         }
 
-        if (objPrincRoles.size() > 0) {
+        if (addPatronSettings(dip.getContentObject())) {
             status.add(FacetConstants.PATRON_SETTINGS);
         }
 
         return status;
     }
 
+    private boolean addPatronSettings(ContentObject contentObj) {
+        List<String> objTypes = contentObj.getTypes();
+        ResourceType objType = ResourceType.getResourceTypeForUris(objTypes);
+        if (objType == null || objType.equals(Cdr.AdminUnit.getURI())) {
+            return false;
+        }
+
+        return hasPatronSettings(contentObj.getPid(), objType);
+    }
+
+    private boolean hasPatronSettings(PID pid, ResourceType rescType) {
+        List<RoleAssignment> objPatronRoles = objAclFactory.getPatronRoleAssignments(pid);
+        int numRoles = objPatronRoles.size();
+        boolean isCollection = rescType.equals(Cdr.Collection.getLocalName());
```

Comment:
you can pass the contentObj in instead of rescType, then do `contentObj instanceof CollectionObject`. Might make sense at some point to add a method to ContentObject to getResourceType or something, which would return the ResourceType. But we probably don't need that here.